src/main.c

```diff
@@ -112,6 +113,7 @@ static void load_port_config(struct uci_context *uci, struct uci_section *s)
 		ULOG_ERR("invalid port id=%lu for %s", id, name);
 		return;
 	}
+	config.port_count = MAX(config.port_count, id);
```

Comment:
I was a little worried about a race condition against a callback after a reload if the port count changes, but I don't think there's reason to worry about it here.

Reply:
This may be asynchronous code, but it is sequential. You're guaranteed to exit `load_port_config()` before you reach other parts of the code that use this config. You don't need to be re-entrant or thread-safe because you're not being pre-empted or context-swapped by some other task.

Reply:
I was thinking something like `ubus call poe info` ... but now I'm realizing how little I know/knew about how ubus is written -- I'm guessing that the consumption of tasks is performed by the daemon exposing the callback through its use of a `uloop`? Rather than there being some kind of shared object code being exposed to the `ubus` binary here.

Reply:
There's one event loop. There's only one function ever getting called at any given time. If you look carefully, you won't see any blocking function. No sleep() or wait_for(). If a function were to block, then we couldn't process events for that amoutn of time. ubus events are no different. In the end you're waiting fir events on a file descriptor or socket.